pkg/kubernetes/portforward.go

```diff
@@ -78,24 +47,13 @@ func (pf *PortForwarder) Start(readyFunc func(pf *PortForwarder) error) error {
 		errChan <- err
 	}()
 
-	// Stop forwarding in case OS signals are received
-	go func() {
-		<-sigChan
-		if pf.stopChan != nil {
-			close(pf.stopChan)
-		}
-	}()
-
```

Comment:
Why are the signals being removed? These were there to ensure that when someone uses the `osm dashboard` command, the command blocks till the user hits `Ctrl + C` to close the port forwarded connection gracefully.

Reply:
The dashboard command already handles the signals explicitly. Seeing as that was the only place responding to those signals, it made sense to remove it here.

Reply:
Got, thanks for clarifying.

Reply:
Though should't we close port forwarded connections generically when using the helpers when OS signals are received?

Reply:
I don't see a need really. Seems like that should be the responsibility of the caller. The port forward will still stop once the process exits even if `Stop()` isn't called.

Reply:
That's true, but its generally better to handle these sort of signals so we know if the channel was specifically closed due to a signal from the OS for some reason.

In this case since its not a big deal given the blocking cli command is handling signals.

Reply:
What I meant to say is the process will exit, but some code executing in a goroutine leveraging the port-forwarding connection could panic if not done gracefully.

Reply:
Would it work to make the `stopChan` available from inside the `readyFunc` so callers can tell when it's been stopped from some other goroutine?

That would allow something like this:

```go
func TestPortForwardStopDuringReadyCallback(t *testing.T) {
	dialer := &fakeDialer{
		conn: &noopConnection{},
	}

	pf, err := NewPortForwarder(dialer, ":80")
	if err != nil {
		t.Fatal("error creating PortForwarder:", err)
	}

	go func() {
		err = pf.Start(func(stop <-chan struct{}) error {
			<-stop
			return nil
		})
		if err != nil {
			t.Error("error running port forward:", err)
		}
	}()

	<-pf.readyChan
	pf.Stop()
}
```

Reply:
I think even leaving this as before should work fine. `dashboard.go` is listening on OS signals for its own business logic (so it can block the command and leave the dashboard open), while `portforward.go` attempts to gracefully close the managed port-forwarded connections. 

Was there a specific reason to remove the existing signalling logic otherwise?

Reply:
Even adding the signal handling back here and removing the `Stop()` call I added in dashboard.go, I don't see the signal being picked up by the PortForwarder when I ctrl+C `osm dashboard`, so it doesn't seem like that was doing anything anyway in that case.

(Sorry for dragging this out, still trying to understand what could go wrong.)

Reply:
Not sure if signals on MacOS are any different, seems suspicious regardless.

Reply:
I got the signal handling to work and sync, added it back.

Reply:
What was the issue before?

Reply:
The `defer signal.Stop(sigChan)` in `Start()` was unregistering the signal handler in the `PortForwarder` once that function returned, so by the time the dashboard command was listening for the signal, the `PortForwarder` wasn't, so no cleanup was being done. I also added a `Done()` method on `PortForwarder` that returns a channel to wait on to ensure cleanup is complete.